web/packages/teleport/src/services/userPreferences/userPreferences.ts

```diff
@@ -50,3 +53,37 @@ export function makeDefaultUserPreferences(): UserPreferences {
     },
   };
 }
+
+function camelCaseToSnakeCase(str: string) {
+  return str.replace(/[A-Z]/g, letter => `_${letter.toLowerCase()}`);
+}
+
+export function createUpdateMask(object: Record<string, any>) {
```

Comment:
Perhaps a short comment that explains the overall transformation taking place here would be nice.